### PR TITLE
Add Golden Retriever docs

### PR DESCRIPTION
**Pull request by arturi**

### CODE REVIEW

File: docs/golden-retriever.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+sidebar_position: 10
+---
+
+# Golden Retriever
+
+The `@uppy/golden-retriever` plugin saves selected files in your browser cache,
+so that if the browser crashes, or the user accidentally closes the tab, Uppy
+can restore everything and continue uploading as if nothing happened. You can
+read more about it
+[on our blog](https://uppy.io/blog/2017/07/golden-retriever/).
```
Comment (aduh95) [inline docs/golden-retriever.mdx:11]: Do we know if the links to the blog would still work without modifications?

File: docs/golden-retriever.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+The Golden Retriever uses three methods of browser data storage:
+
+- `LocalStorage` to store file metadata and Uppy state only
```
Comment (aduh95) [inline docs/golden-retriever.mdx:15]:
```suggestion
- `LocalStorage` to store file metadata and Uppy state only.
```

File: docs/golden-retriever.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+- `IndexedDB` for small files, usually under 5MB
```
Comment (aduh95) [inline docs/golden-retriever.mdx:16]:
```suggestion
- `IndexedDB` for small files, usually under 5MiB.
```

File: docs/golden-retriever.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import Uppy from '@uppy/core';
+import Dashboard from '@uppy/dashboard';
+import Tus from '@uppy/tus';
+import GoldenRetriever from '@uppy/golden-retriever';
+
+new Uppy().use(Dashboard).use(Tus).use(GoldenRetriever);
+```
+
+By default, Golden Retriever will only use the `IndexedDB` storage, which is
+good enough for files up to 5 megabytes (configurable). `Service Worker` is
```
Comment (aduh95) [inline docs/golden-retriever.mdx:80]:
```suggestion
good enough for files up to 5 MiB (configurable). `Service Worker` is
```

File: docs/golden-retriever.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+:::tip For Webpack, check out
+[serviceworker-webpack-plugin](https://github.com/oliviertassinari/serviceworker-webpack-plugin).
+:::
+
+```js
+// sw.js
+
+import('@uppy/golden-retriever/lib/ServiceWorker');
+```
```
Comment (aduh95) [inline docs/golden-retriever.mdx:99]:
````suggestion
1. Bundle your own service worker `sw.js` file with Uppy GoldenRetriever’s
   service worker.

   :::tip

   For Webpack, check out
   [serviceworker-webpack-plugin](https://github.com/oliviertassinari/serviceworker-webpack-plugin).

   :::

   ```js title="sw.js"
   import('@uppy/golden-retriever/lib/ServiceWorker');
   ```
````

File: docs/golden-retriever.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+			console.log(
+				'ServiceWorker registration successful with scope: ',
+				registration.scope,
+			);
+		})
+		.catch((error) => {
+			console.log(`Registration failed with ${error}`);
+		});
+}
+```
```
Comment (aduh95) [inline docs/golden-retriever.mdx:120]:
````suggestion
2. Register it in your app’s entry point:

   ```js
   // you app.js entry point
   uppy.use(GoldenRetriever, { serviceWorker: true });

   if ('serviceWorker' in navigator) {
   	navigator.serviceWorker
   		.register('/sw.js') // path to your bundled service worker with GoldenRetriever service worker
   		.then((registration) => {
   			console.log(
   				'ServiceWorker registration successful with scope: ',
   				registration.scope,
   			);
   		})
   		.catch((error) => {
   			console.log(`Registration failed with ${error}`);
   		});
   }
   ```
````

File: docs/golden-retriever.mdx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+Voilà, that’s it. Happy retrieving!
+
+## API
+
+### Options
+
+#### `id`
+
+A unique identifier for this plugin (`string`, default: `'GoldenRetriever'`)
```
Comment (aduh95) [inline docs/golden-retriever.mdx:130]:
```suggestion
A unique identifier for this plugin (`string`, default: `'GoldenRetriever'`).
```

File: docs/golden-retriever.mdx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+#### `expires`
+
+How long to store metadata and files for. Used for `LocalStorage` and
+`IndexedDB` (`number`, default: `24 * 60 * 60 * 1000` // 24 hours)
```
Comment (aduh95) [inline docs/golden-retriever.mdx:135]:
```suggestion
`IndexedDB` (`number`, default: `24 * 60 * 60 * 1000` // 24 hours).
```

File: docs/golden-retriever.mdx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+#### `serviceWorker`
+
+Whether to enable `Service Worker` storage (`boolean`, default: `false`)
```
Comment (aduh95) [inline docs/golden-retriever.mdx:139]:
```suggestion
Whether to enable `Service Worker` storage (`boolean`, default: `false`).
```

File: docs/golden-retriever.mdx (hunk 1 above)
Comment (arturi) [reply docs/golden-retriever.mdx:11]: I believe that was my aim with the blog move, yes. But we should also add some CI link tester thing, I think we discussed it.

**Pull request merged**